src/main/java/org/spongepowered/api/service/permission/Subject.java

```diff
@@ -113,12 +112,15 @@
 
     /**
      * Test whether the subject is permitted to perform an action given as the
-     * given permission string.
+     * given permission string. This must return the same value as
+     * {@link #hasPermission(Set, String)} using {@link #getActiveContexts()}.
      *
      * @param permission The permission string
      * @return True if permission is granted
      */
-    boolean hasPermission(String permission);
+    default boolean hasPermission(String permission) {
```

Comment:
These are @zml2008's changes. It's displayed in the diff because I branched off of bleeding, merged master into my new branch, and then applied my changes.